controllers/networkchaos/types.go

```diff
@@ -15,15 +15,22 @@ package networkchaos
 
 import (
 	"context"
+	"fmt"
 
-	"github.com/go-logr/logr"
+	"github.com/pingcap/chaos-mesh/controllers/common"
 
-	"github.com/pingcap/chaos-mesh/api/v1alpha1"
-	"github.com/pingcap/chaos-mesh/controllers/networkchaos/netem"
-	"github.com/pingcap/chaos-mesh/controllers/networkchaos/partition"
+	"github.com/pingcap/chaos-mesh/controllers/twophase"
+
+	"github.com/go-logr/logr"
 
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/pingcap/chaos-mesh/api/v1alpha1"
+	commonNetem "github.com/pingcap/chaos-mesh/controllers/common/networkchaos/netem"
```

Comment:
I find the common and twwphase wrapping and renaming a bit duplicated. Is it possible to eliminate them? I am not very familiar with the code base ... it seems we have two set of interfaces under common and twophase. And the wrappers are almost same (except for the last parameter)

```go
// under common/types
// InnerCommonReconcile used in common chaos reconcile
type InnerCommonReconcile interface {
	Apply(ctx context.Context, req ctrl.Request, chaos InnerCommonObject) error
	Recover(ctx context.Context, req ctrl.Request, chaos InnerCommonObject) error
	Object() InnerCommonObject
}

// under twophase/types
type InnerReconciler interface {
	Apply(ctx context.Context, req ctrl.Request, chaos InnerObject) error
	Recover(ctx context.Context, req ctrl.Request, chaos InnerObject) error
	Object() InnerObject
}

// under common/podfailure
// CommonReconciler is reconciler for podfailure
type CommonReconciler struct {
	*podfailure.Reconciler
}

// Apply would perform common chaos for podchaos
func (r *CommonReconciler) Apply(ctx context.Context, req ctrl.Request, chaos common.InnerCommonObject) error {
	return r.Perform(ctx, req, chaos)
}

// under twophase/podfailure
// TwoPhaseReconciler reconcile the networkchaos
type TwoPhaseReconciler struct {
	*podfailure.Reconciler
}

// Apply implement InnerReconciler.Apply
func (r *TwoPhaseReconciler) Apply(ctx context.Context, req ctrl.Request, chaos twophase.InnerObject) error {
	return r.Perform(ctx, req, chaos)
}
```

Instead of having a layout where we duplicate package names under common and twophase

```
- podfailure
- netem
- common
   - podfailure
   - netem
- twophase
   - podfaillure
   - netem
```

We can define those wrappers in `podfailure` and `netem` directly with new factory func like `NewCommonReconciler`, to avoid having two extra packages for each chaos and use `netem.NewCommonReconciler` instead of `commonNetem.NewReconciler`. This removes subpackage from `common` and `twophase` packages and solves the renaming problem. 

```go
// podfailure

// NewReconciler would create new reconciler for podfailure chaos
func NewReconciler(c client.Client, log logr.Logger, req ctrl.Request) *Reconciler {
	return &Reconciler{Client: c, Log:  log}
}

func NewCommonReconciler(...) *CommonReconciler {
}

func NewTwoPhaseReconciler(...) *TwoPhaseReconciler {
}
```

btw: the wrappers should be generated

Reply:
Deeply agree with you.

Reply:
fixed in #201